CRM/Civicase/Hook/SelectWhereClause/LimitCaseQueryToAccessibleCaseCategories.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Restrict case query result to accessible case categories if case entity is involved.
+ */
+class CRM_Civicase_Hook_SelectWhereClause_LimitCaseQueryToAccessibleCaseCategories {
+
+  /**
+   * Restrict search to accessible case categories if case entity is involved.
+   *
+   * @param string $entity
+   *   Entity name.
+   * @param array $clauses
+   *   Where clauses.
+   */
+  public function run($entity, array &$clauses) {
+
+    if (!$this->shouldRun($entity)) {
+      return;
+    }
+    $accessibleCaseCategories = CaseCategoryHelper::getAccessibleCaseTypeCategories();
```

Comment:
For administrators or users that have access to all case categories, we don't need to add the clause at all. We can use this function to avoid that: https://github.com/compucorp/uk.co.compucorp.civicase/blob/50aa75dde19e1828b13cb6614795809e105adcad/CRM/Civicase/BAO/Query/CaseCategory.php#L75-L80